Full feature compatibility with WPF's Colors class

adrianknight89 opened the pull request:
### Description of Change ###

WPF's `Colors` class has 141 colors from UNIX X11 color table. See https://msdn.microsoft.com/en-us/library/system.windows.media.colors(v=vs.110).aspx

XF had 19 colors, so I added the remaining for full compatibility.

**Questions**

- XF's definition of Pink is wrong. It should be `rgb(255, 192, 203)`. If we want to change it, it'll be a breaking change. How should this be handled? (Also XF Transparent is black with 0 alpha whereas .NET counterpart is white with 0 alpha. At least this doesn't represent a bug like that of Pink.)
- I added hex values of the colors since msdn shows hex in its own documentation as well as the color table. Should the original 19 colors also show hex? See `Color.cs`
- Original documentation is wrong in that it's using the wrong word RGB instead of HEX. For example: `White, the color that is represented by the RGB value #ffffff.` Should I fix this?

P.S. I automated most of the work, so there shouldn't be an error with the hex values.

dnfclas commented:
Hi __@adrianknight89__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

StephaneDelcroix commented:
before fixing the docs and everything, wait for @jassmith opinion on whether or not we will include this.

StephaneDelcroix commented on Xamarin.Forms.Core/Color.cs:
```diff
@@ -371,6 +371,130 @@ public static Color FromHsla(double h, double s, double l, double a = 1d)
 		public static readonly Color White = FromRgb(255, 255, 255);
 		public static readonly Color Yellow = FromRgb(255, 255, 0);
 
+		// remaining colors in WPF's System.Windows.Media.Colors
```
but actually, I do not care

adrianknight89 commented:
I already provided the docs. GitHub won't show it because it's too big. You should be able to see it in your IDE.

StephaneDelcroix commented:
my wrong, they are indeed at the bottom of the changes.

StephaneDelcroix commented:
@adrianknight89 we'll take this change in, including the change in the Pink Color. Could you update the PR ?

thanks

adrianknight89 commented on Xamarin.Forms.Core/Color.cs:
```diff
@@ -371,6 +371,130 @@ public static Color FromHsla(double h, double s, double l, double a = 1d)
+		public static readonly Color Violet = FromHex("#FFEE82EE");
+		public static readonly Color Wheat = FromHex("#FFF5DEB3");
+		public static readonly Color WhiteSmoke = FromHex("#FFF5F5F5");
+		public static readonly Color YellowGreen = FromHex("#FF9ACD32");
```
@StephaneDelcroix Should the docs be sorted as well?

StephaneDelcroix replied:
I don't think so. That's the responsibility of the displaying tool

adrianknight89 replied:
Sorted colors alphabetically. Same was done in `ColorTypeConverter.cs`

adrianknight89 commented on Xamarin.Forms.Core/Color.cs:
```diff
@@ -371,6 +371,130 @@ public static Color FromHsla(double h, double s, double l, double a = 1d)
 		public static readonly Color Yellow = FromRgb(255, 255, 0);
 
+		// remaining colors in WPF's System.Windows.Media.Colors
+		public static readonly Color AliceBlue = FromHex("#FFF0F8FF");
```
Changed to `FromRgb()`.

adrianknight89 commented:
I updated Pink to reflect the correct value. I also updated Transparent to be transparent from white instead of black. The end result should be the same since the alpha is 0.

Should we remove the obsolete color? It was set to obsolete in 1.3. I think it's time to get rid of it and let devs update to the correct spelling.

StephaneDelcroix commented:
@adrianknight89 unfortunately we can't remove API, even the ones Obsoleted for a long time.
